zeebe/restore/src/main/java/io/camunda/zeebe/restore/PartitionRestoreService.java

```diff
@@ -176,7 +187,10 @@ private void moveSnapshotFiles(final Backup backup) {
 
     @SuppressWarnings("resource")
     final RestorableSnapshotStore snapshotStore =
-        new FileBasedSnapshotStore(partition.id().id(), partition.dataDirectory().toPath());
+        new FileBasedSnapshotStore(
+            partition.id().id(),
+            partition.dataDirectory().toPath(),
+            provider != null ? provider : new ChecksumProviderRocksDBImpl());
```

Comment:
:wrench: Let's do this fallback to `ChecksumProviderRocksDBImpl` in the constructors.